### ISSUE TITLE
NaNs in G-cases with pgigpu

### ISSUE THREAD

**Issue by amametjanov** [E3SM-Project/E3SM#5042]
Repo: `maint-2.0` from June 2, hash af7e689913b22158e8c86705341fb04e49f85df0
Case: `./cime/scripts/create_test --compiler pgigpu PFS_P1024.T62_oRRS18to6v3.GMPAS-IAF.summit_pgigpu.bench-gmpas_noio`

Run progresses to 
```
 0001-01-01_14:00:00  WC time:           460.749
 === Completed coupling interval in ocn_run_mct. ===
 0001-01-01_14:06:00  WC time:           468.332
 0001-01-01_14:12:00  WC time:           469.574
 0001-01-01_14:18:00  WC time:           471.008
```
and aborts
```
1: 512: --------------------------------------------------------------------------
1: 512: MPI_ABORT was invoked on rank 512 in communicator MPI_COMM_WORLD
1: 512: with errorcode 1405682704.
1: 512:
1: 512: NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
1: 512: You may or may not see output from other processes, depending on
1: 512: exactly when Open MPI kills them.
1: 512: --------------------------------------------------------------------------
```
with
```
CRITICAL ERROR: ERROR: State validation failed. See block stats files for more information.
```
with block stats
```
 ERROR: NaN Detected in state see below for which field contained a NaN.
   -- Statistics information for block fields
     Field: latCell
         Min:     1.191280815377864
         Max:     1.275941419532617
     Field: lonCell
         Min:    0.9663009153186918
         Max:     1.204136975475011
     Field: xCell
         Min:     708253.2824618289
         Max:     1137362.658824743
     Field: yCell
         Min:     1544404.576900129
         Max:     2193588.837771315
     Field: zCell
         Min:     5917879.315745871
         Max:     6096274.257709019
     Field: areaCell
         Min:     36673637.04579713
         Max:     38492077.85244904

 ERROR: NaN Detected in normalVelocity.
   -- Statistics information for normalVelocity fields
     Field: seaIcePressure
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: atmosphericPressure
         Min:     100487.1689482392
         Max:     100887.4387269987
     Field: windStressZonal
         Min:   -4.5022742092250823E-002
         Max:   -1.5972184783098579E-002
     Field: windStressMeridional
         Min:   -5.4865903088738696E-003
         Max:    1.4073500894236932E-002
     Field: surfaceStressMagnitude
         Min:    1.0445534278135863E-002
         Max:    4.1469374778702973E-002
     Field: surfaceStress
         Min:   -4.4097256267437275E-002
         Max:    3.9980498432890837E-002
     Field: angleEdge
         Min:    -3.141592653589793
         Max:     3.141592653589793

 ERROR: NaN Detected in activeTracers.
   -- Statistics information for activeTracers fields
     Field: latentHeatFlux
         Min:    -75.13526745307371
         Max:    -38.86466042541282
     Field: sensibleHeatFlux
         Min:    -220.9362289100335
         Max:    -85.76319378993975
     Field: longWaveHeatFluxUp
         Min:    -311.3172228290222
         Max:    -307.7417996470279
     Field: longWaveHeatFluxDown
         Min:     204.2450604762500
         Max:     215.6938643275701
     Field: seaIceHeatFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: icebergHeatFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: seaIceFreshWaterFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: icebergFreshWaterFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: rainFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: snowFlux
         Min:    1.0485041482582745E-005
         Max:    1.6136168044549781E-005
     Field: iceRunoffFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: riverRunoffFlux
         Min:     0.000000000000000
         Max:    5.3100070586483710E-005
     Field: seaIceSalinityFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
     Field: shortWaveHeatFlux
         Min:     0.000000000000000
         Max:     0.000000000000000
```

I'm wondering if a fix went into `master`, but not yet to `maint-2.0`. Other partitions, besides 1024 (like 2560, 5200), all stop at `0001-01-01_14:18:00`.

**Comment by rljacob** [comment]
Do we want maint-2.0 to work with gpus?  I don't think that's required.

**Comment by amametjanov** [comment]
Yes, it's needed for benchmarking CPU-vs-GPU in v2 (as baseline speedup).

**Comment by amametjanov** [comment]
Just to cross-link the issues: this is for `maint-2.0`, `master` has a different build issue E3SM-Project/E3SM#4903

**Comment by amametjanov** [comment]
Fixed by #5101
